RUFAS/output_manager.py

```
@@ -803,22 +808,52 @@ def save_results(
                 )
                 if exclude_info_maps:
                     filtered_pool = self._exclude_info_maps(filtered_pool)
-                self._route_save_functions(
-                    filter_file,
-                    save_path,
-                    filtered_pool,
-                    produce_graphics,
-                    filter_content,
-                    graphics_dir,
-                )
+
+                if filter_file.startswith(
+                    self.__supported_filter_types_prefixes["report"]
+                ):
+                    self.add_log(
+                        "init_report_generation", "Report Generation Started", info_map
+                    )
+                    report_generator = ReportGenerator()
+                    try:
+                        report_name = filter_content.get(
+                            "name", f"untitled_{self._get_timestamp(True)}"
+                        )
+                        reports[
+                            report_name
+                            if report_name not in reports.keys()
+                            else f"{report_name} {self._get_timestamp(True)}"
+                        ] = {
+                            "values": report_generator.generate_report(
+                                filtered_pool,
+                                filter_content,
+                            )
+                        }
+                    except (ValueError, KeyError) as e:
+                        self.add_error("report generation error", str(e), info_map)
```

Should we raise the exception as we're doing with others (e.g. on lines 915-916 within `_save_variables_to_csv_files()`)?

---

If we raise an error here, then the rest of the filter file which might be intact won't be processed. I think it is better to give the user as much as we can instead of panicking on the first error.